sympy/polys/ring_series.py

```diff
@@ -753,7 +753,7 @@ def rs_diff(p, x):
     for expv in p:
         if expv[n]:
             e = monomial_ldiv(expv, mn)
-            p1[e] = p[expv]*expv[n]
+            p1[e] = R.domain_new(p[expv]*expv[n])
```

Comment:
Thanks for the explanation. The question is whether this conversion should happen in the low level functions like `rs_diff`, or rather if, as I thought, those functions should already assume things have the proper type, so that they don't need to keep typing things all over the place.